### ISSUE TITLE
Countries of South America?

### ISSUE THREAD

**Issue by Esteban82** [GenericMappingTools/dcw-gmt#14]
I run the following code for South America (-R=SA). 

I see that in [dcw-countries.txt ](https://github.com/GenericMappingTools/dcw-gmt/blob/master/dcw-countries.txt) there are many countries tag as SA that don't belong  think we should revised which countries (like Bahamas and Bermuda). 

Maybe we could also created another region like Central America, Caribbean or both. 

According to Wikipedia, this regions belong to North America. I think it would be better to created those regions.

`gmt coast -R=SA -Bagf -W -N1 -png Mapa -JM20c -Sdodgerblue2`

![Mapa](https://user-images.githubusercontent.com/17908951/132404579-eaff51ad-e89b-4c42-a660-782919e93696.png)

**Comment by PaulWessel** [comment]
Interesting.  Yes, I agree we should follow convention here and those are indeed listed under North America.  We should also check that there are not other issues for other continent lists.  I think the steps needed are to

1. Revise the dcw-countries.txt file
2. Move the corresponding country files from SA to NA (and similar if there are other issues for other continents)
3. Update dcw version number and build 2.0.1 version

Looks like all of these should be moved, although a few are not listed in either SA or NA on the wiki but all are listed in the separate Caribbean section.

```
SA AG Antigua
SA AI Anguilla
SA AW Aruba
SA BB Barbados
SA BL Saint Barthélemy
SA BM Bermuda
SA BQ Bonaire, Sint Eustatius and Saba
SA BS Bahamas
SA BZ Belize
SA CR Costa Rica
SA CU Cuba
SA CW Curaçao
SA DM Dominica
SA DO Dominican Republic
SA GD Grenada
SA GP Guadeloupe
SA GT Guatemala
SA HN Honduras
SA HT Haiti
SA JM Jamaica
SA KN Saint Kitts and Nevis
SA KY Cayman Islands
SA LC Saint Lucia
SA MF Saint Martin (French part)
SA MQ Martinique
SA MS Montserrat
SA NI Nicaragua
SA PA Panama
SA PM Saint Pierre and Miquelon
SA PR Puerto Rico
SA SV El Salvador
SA SX Sint Maarten
SA TC Turks and Caicos
SA TT Trinidad
SA VC Saint Vincent and the Grenadines
SA VG British Virgin Islands
SA VI Virgin Islands
```

Alternatively, we make a CB "continent" collection of all of them.

**Comment by PaulWessel** [comment]
I think adding CB makes more sense since if we move them to NA then users of =NA will be just as unhappy as you.  If they really want to include them them can use =NA,=CB

**Comment by Esteban82** [comment]
Ok. I can do it. Later I will check the countries and propose changes for the other continents as well.

**Comment by Esteban82** [comment]
For the Americas, I think it is better to create Caribbean (CB) and Central America (CA). These are the maps that can be created with my modified version of  dcw-countries.txt.

I moved some countries/territories. I add Greenland to NA. And for SA I remove South Georgia and the South Sandwich Islands because they are far from the continent. If some would like to include then, they can use =SA,GS.  I think that the results are pretty good. 

Tomorrow I will add the other continents. I think that maybe we could split Asia in sub-regions? What do you think?

![Caribbean](https://user-images.githubusercontent.com/17908951/132446654-74154019-2eaa-42ee-99b0-98e7c459710c.png)
![CentralAmerica](https://user-images.githubusercontent.com/17908951/132446657-b87bb5b3-6c12-42c8-a646-397ede351ebd.png)
![NorthAmerica](https://user-images.githubusercontent.com/17908951/132446659-757d77c4-ae10-49f3-a3f2-bd98ee93185f.png)
![SouthAmerica](https://user-images.githubusercontent.com/17908951/132446662-932ebac4-184b-4515-9d05-2c80aff22c8d.png)

**Comment by PaulWessel** [comment]
OK on Caribbean and Central America.  Not sold yet on Greenland though since it is sort of Danish and thus Europe...

**Comment by Esteban82** [comment]
## The issue on intercontinental countries:

Yes, Greenland is part of Denmark but geographically is part of North American. It is not that simple, there are intercontinental countries that can't be assigned to one continent (like Russia, Denmark, Greenland, Spain, France, Turkey, US). 

1) For some countries, like France and its [overseas territories](https://upload.wikimedia.org/wikipedia/commons/thumb/5/5f/France_in_the_World_%28%2BAntarctica_claims%29.svg/1280px-France_in_the_World_%28%2BAntarctica_claims%29.svg.png) it is easy to solve because those territory are separated to the European part. I think Greenland falls in this category.

2) In other cases (like Turkey, Hawaii) from a practical point of view it doesn't matter because the map generated includes those area anyway. 

3) For some countries (e.g. Russia, Spain, Portugal) maybe it would be good to split them in different polygons like in the French case. To be clear, we could spit the Canary Islands from the European part of Spain. We probably should include a note to avoid any political comment. 

4) There are some countries/territories that are island in the Indic and Atlantic that I remove from the continent (to avoid the map extends very much from the geographic continent (like the South Sandwich Islands).

**Comment by Esteban82** [comment]
I just see in the [readme](https://github.com/GenericMappingTools/dcw-gmt#adding-new-country-or-state-boundaries), that the countries are placed in folders of the belonging continent (ZZ/YY). 

So, what would happened with the item 4 (remove island from continents), can be any issue? Or should I create some new "continents", like "Artic, Atlantic, Indic" for the Islands that are far from the continents?

**Comment by PaulWessel** [comment]
There will have to be some changes to the gmt_dcw.[ch] codes to work with what I think we will call vresion 2.1.0, not 2.0.1.  But yes, if you wish to move a file from continent XX to YY then I think all is well as long as the  dcw-countries.txt file is updated also.  I think for states we only have the largest 8 countries and none of those are likely to move, right?

So given how these files are used I agree we can make new "continents".  There is Oceania already for the Pacific region I think.  I think we discussed Caribbean (CB) and Central America (CA). You can add additional one along those lines.  We aren't able to really test until those changes are all done in a WIP branch that will require a compatible WIP branch on the GMT side to test things, including a check for DCW version.  

I do not see any need for changes to the DCW2singleshortnc.sh that makes the netCDF file but we will find out.

**Comment by Esteban82** [comment]
>  I think for states we only have the largest 8 countries and none of those are likely to move, right?

Yes, I have to revised/updated the Argentinian states.

The following countries are Kazakhstan, Algeria, D.R of Congo. I don't know if there any many users that studied those countries. Hope 

> So given how these files are used I agree we can make new "continents". 

Ok. Yes, I think the all the Pacific islands are included in Oceania. I will check.

**Comment by Esteban82** [comment]
I found that the following territories are both in dcw-countries.txt and in dcw-states.txt. So we have duplicated data. Is this on purpose? I think we could save some space by removing them from one the lists. What do you think?

US AS American Samoa
US GU Guam
US MP Northern Mariana Islands
US PR Puerto Rico
US UM United States Minor Outlying Islands
US VI Virgin Islands, U.S.

**Comment by PaulWessel** [comment]
Maybe it was set up so that

1. **-R**US only gives the 50 states of the US, not extending that region to Guam and SA
2. Still retrain the US affinity (they are not independent)

I think (1) is important, while as for (2) I think nobody is going to use +L to look for such territories.  Consider France: It has French Polynesia but since we dong list states (compartments) for France they do not show up with **+L**.  So to be consistent, I agree we should not let any of the 8 large countries with states include these "territories" if they are also included as separate.

**Comment by Esteban82** [comment]
Ok. So I will delete them from the US folder in the PR.

**Comment by Esteban82** [comment]
I fixed all the bugs that I found. I will test the data tomorrow.

The only thing remaining is what should I do with the "continents".  As I said somewhere else, I think here we could keep the continents that we already have (more based on "political/cultural"). And leave the continents based on geographic for the new feature request. 

 So in this case, here we just would need to reassigned some countries (like many of SA to NA). This way, the docs wouldn't need to be update with the new continents (Caribbean, Central America).

**Comment by PaulWessel** [comment]
Yes, that is a good plan; thanks. We could declare these as bug fixes and build 2.0.1 and update that asap.  New features may take a bit longer to finalize but when they do we can release 2.1.0.  It does not matter that the time between these is relatively short, as long as they are actual releases that will be found if someone desperately wants to avoid 2.1.0.
